File: frontend/src/types/subcontractor.ts
export interface Subcontractor {
  id: string;
  company_name: string;
  trade: string;
  trades: string[];
  city: string;
  state: string;
  service_radius_miles: number;
  contact_name: string;
  email: string;
  phone: string;
  hourly_rate: number | null;
  project_rate: number | null;
  available_from: string;
  available_to: string;
  booked_weeks?: Array<{ project: string; start_week: number; end_week: number }>;
  project_types: string[];
  rating: number;
}

export interface PromptTemplate {
  id: string;
  slug: string;
  name: string;
  description: string;
  category: string;
  template_text: string;
  variables: string[];
  model: string;
  max_tokens: number;
  version: number;
  is_default: boolean;
}

File: backend/app/models/subcontractor.py
from pydantic import BaseModel
from typing import Optional


class BookedWeek(BaseModel):
    project: str
    start_week: int
    end_week: int


class SubcontractorCreate(BaseModel):
    company_name: str
    trade: str
    trades: list[str] = []
    city: str
    state: str
    service_radius_miles: float = 50.0
    contact_name: str = ""
    email: str = ""
    phone: str = ""
    hourly_rate: Optional[float] = None
    project_rate: Optional[float] = None
    available_from: Optional[str] = None
    available_to: Optional[str] = None
    booked_weeks: list[BookedWeek] = []
    project_types: list[str] = []
    rating: float = 3.0


class SubcontractorResponse(SubcontractorCreate):
    id: str


class MatchRequest(BaseModel):
    bid_id: str
    location: Optional[str] = None
    trades: Optional[list[str]] = None
